app/models/rook.rb

```diff
@@ -1,4 +1,18 @@
 class Rook < Piece
   scope :piece, -> {where(type: 'Rook')}
 
+  def valid_move?(position_x, position_y)
+    self.position_x = current_x
+    self.position_y = current_y
+    position_x = x_destination
+    position_y = y_destination
+
+    if current_x == x_destination && is_obstructed_vertically? == false # vertical move
```

Comment:
we could refactor it into two methods:

`valid_vertical_move(position_x, position_y)`and `valid_horizontal_move(position_x, position_y)` and put the logic in there

Reply:
@ilya done!